### PR TITLE
Upgrade sinon to version 4

### PR DESCRIPTION
**Pull request by eatyourgreens**
Upgrades sinon from 1.1 to the latest version.

Fixes broken `stub()` calls and removes deprecated `reset()` calls.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/pages/project/workflow-selection.spec.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ const projectRoles = [
 function mockPanoptesResource(type, options) {
   const resource = apiClient.type(type).create(options);
   apiClient._typesCache = {};
-  sinon.stub(resource, 'save', () => Promise.resolve(resource));
+  sinon.stub(resource, 'save').callsFake(() => Promise.resolve(resource));
```
Comment (simoneduca) [inline app/pages/project/workflow-selection.spec.js]: This is broken. Looking into it now.

**Comment by simoneduca** [comment]
Did you try running the tests after your changes? Tests are broken for me. I'm getting `_get__(...).stub(...).callsFake is not a function`, so it's probably just some syntax.

**Comment by eatyourgreens** [comment]
`All checks have passed` according to Travis.

I ran `npm test` before creating this PR, yes.

From what I can tell, you're still running sinon version 1.

**Comment by simoneduca** [comment]
Ah, my bad sorry! All looking good to me.

**Pull request merged**